homeassistant/components/goodwe/sensor.py

```diff
@@ -0,0 +1,262 @@
+            # do not include unknown sensors
+            continue
+        entities.append(InverterSensor(coordinator, device_info, inverter, sensor))
+
+    async_add_entities(entities)
+
+    # Add services
+    platform = entity_platform.async_get_current_platform()
+    platform.async_register_entity_service(
+        SERVICE_SET_WORK_MODE,
+        {vol.Required(ATTR_WORK_MODE): vol.Coerce(int)},
+        "async_set_work_mode",
+    )
+    platform.async_register_entity_service(
+        SERVICE_SET_ONGRID_BATTERY_DOD,
+        {vol.Required(ATTR_ONGRID_BATTERY_DOD): vol.Coerce(int)},
+        "async_set_ongrid_battery_dod",
+    )
+    platform.async_register_entity_service(
+        SERVICE_SET_GRID_EXPORT_LIMIT,
+        {vol.Required(ATTR_GRID_EXPORT_LIMIT): vol.Coerce(int)},
+        "async_set_grid_export_limit",
+    )
+
+    return True
+
+
+class InverterEntity(CoordinatorEntity, SensorEntity):
+    """Entity representing the inverter instance itself."""
+
+    _MAIN_ENTITY_SENSOR = "ppv"
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        device_info: DeviceInfo,
+        inverter: Inverter,
+    ) -> None:
+        """Initialize the main inverter entity."""
+        super().__init__(coordinator)
+        self.entity_id = f".{DOMAIN}_inverter"
+        self._attr_unique_id = f"{DOMAIN}-{inverter.serial_number}"
+        self._attr_device_info = device_info
+        self._attr_icon = "mdi:solar-power"
+        self._attr_name = "PV Inverter"
+        self._attr_native_unit_of_measurement = POWER_WATT
+        self._attr_extra_state_attributes = {"serial_number": inverter.serial_number}
+        self._inverter: Inverter = inverter
+
+    async def async_set_work_mode(self, work_mode: int) -> None:
+        """Set the inverter work mode."""
+        await self._inverter.set_work_mode(work_mode)
+
+    async def async_set_ongrid_battery_dod(self, ongrid_battery_dod: int) -> None:
+        """Set the on-grid battery dod."""
+        await self._inverter.set_ongrid_battery_dod(ongrid_battery_dod)
+
+    async def async_set_grid_export_limit(self, grid_export_limit: int) -> None:
+        """Set the grid export limit."""
+        await self._inverter.set_grid_export_limit(grid_export_limit)
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self.coordinator.data is not None
+
+    @property
+    def native_value(self):
+        """Return the value reported by the sensor."""
+        if self.coordinator.data is not None:
+            new_value = self.coordinator.data.get(self._MAIN_ENTITY_SENSOR)
+            # If no new value was provided, keep the previous
+            if new_value is not None:
+                self._attr_native_value = new_value
+
+        return self._attr_native_value
+
+    @property
+    def state_attributes(self):
+        """Return the inverter state attributes."""
+        return self.coordinator.data
```

Comment:
I believe this feature is appreciated by many users. There are lots of sensors exposed by the inverter and people many times do not want to show them in their UIs as individual sensors.
By having these exposed as state attributes of the "main" sensor, one can simply just add single badge to the UI and when a user wants to get overview of the inverter and its sensor values, it can get a very convenient view over all of them.
It may sound unusual, but I'd really like to keep this feature.

Reply:
State attributes will make the database bigger. And state attributes cannot be disabled or ignored.

Reply:
I have scan interval set to 10s and running the integration for more than a year and I did not observe any problems with database size. If the space is important for someone, he can exclude this sensor from recorder. (It's only this single sensor which has these state attributes, rest of the sensors have none state attributes.)

Reply:
Ok, that makes sense.

Reply:
I think this may be rejected by "master" reviewers, but maybe they will be satisfied with your explaination.

Reply:
Some users will miss that feature, but ok, will remove it.